components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/opensearchs3bucketaccesschecktrigger/opensearchs3bucketaccesscheck_test.go

```diff
@@ -0,0 +1,258 @@
+			  "resolution_msg": "Setup OpenSearch with valid configurations for S3 backup"
+			}
+		  ]
+		}
+	  }
+	`
+
+	apiTriggerResponseSuccess = `
+	[
+		{
+			"status": "SUCCESS",
+			"host": "open-search-url",
+			"node_type": "opensearch",
+			"result": {
+				"passed": true,
+				"checks": [
+					{
+						"title": "Create test backup",
+						"passed": true,
+						"success_msg": "OpenSearch is able to create backup to provided S3",
+						"error_msg": "",
+						"resolution_msg": ""
+					}
+				]
+			}
+		}
+	]
+ `
+
+	apiTriggerResponseFailure = `[
+		{
+		"status": "SUCCESS",
+		"host": "open-search-url",
+		"node_type": "opensearch",
+		"result": {
+			"passed": false,
+			"checks": [
+				{
+					"title": "Create test backup",
+					"passed": false,
+					"success_msg": "",
+					"error_msg": "OpenSearch is not able to create backup to provided S3",
+					"resolution_msg": "Setup OpenSearch with valid configurations for S3 backup"
+				}
+			]
+		}
+	}
+	]
+ `
+)
+
+func TestOpensearchS3BucketAccessCheck_Run(t *testing.T) {
+
+	type args struct {
+		config models.Config
+	}
+	tests := []struct {
+		name             string
+		args             args
+		httpResponseCode int
+		isPassed         bool
+		response         string
+		isError          bool
+	}{
+		{
+			name: "Success Response",
+			args: args{
+				config: models.Config{
+					ExternalOS: externalOs,
+					Backup: models.Backup{
+						ObjectStorage: s3Properties,
+					},
+				},
+			},
+			httpResponseCode: http.StatusOK,
+			isPassed:         true,
+			response:         apiTriggerResponseSuccess,
+			isError:          false,
+		},
+		{
+			name: "Failure Response",
+			args: args{
+				config: models.Config{
+					ExternalOS: externalOs,
+					Backup: models.Backup{
+						ObjectStorage: s3Properties,
+					},
+				},
+			},
+			httpResponseCode: http.StatusOK,
+			isPassed:         false,
+			response:         apiTriggerResponseFailure,
+			isError:          false,
+		},
+		{
+			name: "Internal Server Error",
+			args: args{
+				config: models.Config{
+					ExternalOS: externalOs,
+					Backup: models.Backup{
+						ObjectStorage: s3Properties,
+					},
+				},
+			},
+			httpResponseCode: http.StatusInternalServerError,
+			isPassed:         false,
+			response:         "error while connecting to the endpoint, received invalid status code",
+			isError:          true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var want []models.CheckTriggerResponse
+			json.Unmarshal([]byte(tt.response), &want)
+			server, host, port := createDummyServer(t, tt.httpResponseCode, tt.isPassed)
+			defer server.Close()
+
+			osb := NewOpensearchS3BucketAccessCheck(logger.NewLogrusStandardLogger(), port)
+			osb.host = host
+
+			got := osb.Run(tt.args.config)
+			if tt.isError {
+				assert.Len(t, got, 1)
+				assert.NotNil(t, got[0].Result.Error.Error)
+				assert.Equal(t, "opensearch", got[0].NodeType)
```

Comment:
you need to use the constant here instead of `opensearch`

Reply:
done